skhep/math/vectors.py

```diff
@@ -3,1029 +3,394 @@
-        Multiplication of two vectors is equivalent to the dot product, see dot(...).
-        Example:
-        >>> v2 = v1 * 2
-        >>> number = v1 * v3
-        """
-        if isinstance ( other ,  Vector3D ) :
-            return self.dot(other)
-        v = self.copy()
-        v *= other
-        return v
-        
-    def __rmul__(self, other):
-        """Right multiplication of the vector by either another vector or a number."""
-        return self.__mul__(other)
-                
-    def __truediv__(self, number):
-        """Division of the vector by a number."""
-        v = self.copy()
-        v /= number
-        return v
-
-    __div__ = __truediv__
-        
-    def __eq__  ( self , other ) :
-        """Equality to another vector, or, equality to zero
-        :Example:
-        >>> v1 = ...
-        >>> v2 = ...
-        >>> print v1 == v2
-        >>> print v1 == 0  
-        """
-        from skhep.math.numeric import isequal
-
-        ## comparsion with scalar zero, very useful  in practice
-        if isinstance ( other , ( float , int , long ) ) and isequal ( other , 0 ) : 
-            return isequal ( self[0] , 0 ) and isequal ( self[1] , 0 ) and isequal ( self[2] , 0 ) 
-        elif not isinstance ( other , Vector3D) :
-            return NotImplemented
-        ##
-        return isequal ( self[0] , other[0] ) and isequal ( self[1] , other[1] ) and isequal ( self[2] , other[2] )  
-
-    def __ne__  (self, other) :
-        """Non-equality to another vector
-        :Example:
-        >>> v1 = ...
-        >>> v2 = ...
-        >>> print v1 != v2 
-        """
-        return not ( self == other )
-
-    def __nonzero__ ( self ) :
-        """Nonzero  vector?
-        :Example:
-        >>> vct = ...
-        >>> if vct : ...
-        """
-        return 0 != self.mag2
-
-    __bool__ = __nonzero__
-    
-    def __iter__(self):
-        """Iterator implementation for the vector components."""
-        return self.__values.__iter__()
-
-    def dot(self, other):
-        """Dot product with another vector."""
-        return sum(v1 * v2 for v1, v2 in zip(self.__values, other.__values))
-
-    def cross(self, other ):
-        """Cross product with another vector."""
-        return Vector3D( self[1] * other[2] - self[2] * other[1],
-                         self[2] * other[0] - self[0] * other[2],
-                         self[0] * other[1] - self[1] * other[0]
-                        )
-                        
-    def rotate(self, angle, *args):
-        """Rotate vector by a given angle (in radians) around a given axis."""
-        if len(args) == 1 and isinstance(args[0], Vector3D):
-            ux, uy, uz = args[0].__values
-        elif len(args) == 1 and len(args[0]) == 3:
-            ux, uy, uz = args[0]
-        elif len(args) == 3:
-            ux, uy, uz = args
+            return self / self.mag()
+
+
+    @property
```

Comment:
Here does it mean a Vector2D can have a T attribute?

Reply:
T is transpose, from numpy, not `t` (time). Transposing makes them normal numpy arrays.

Reply:
PS: T is already available, I'm just fixing it to return normal numpy arrays, since a non-XD XD vector is not a good idea...